src/codegen.cpp

```diff
@@ -7159,10 +7159,11 @@ extern "C" void jl_init_codegen(void)
     initializeAnalysis(Registry);
 #endif
 
-    InitializeNativeTarget();
-    InitializeNativeTargetAsmPrinter();
-    InitializeNativeTargetAsmParser();
-    InitializeNativeTargetDisassembler();
+    InitializeAllTargets();
```

Comment:
Could you please elaborate ?

To elaborate on what I said,
1. LLVM was built to target only X86 and NVPTX. So, libLLVM-3.9.so didn't contain any function pertaining to other back-ends.
2. libjulia.so, built with src/codegen.cpp containing calls to IntializeAll*(), also didn't contain calls to any functions pertaining to backends other than X86 and NVPTX.

So, using IntializeAll*() helps in,
1. initialising just the back-ends that the LLVM build targeted 
2. without using any kind of conditionals.